cpp/path_module/algorithm/path.cpp

```diff
@@ -1,301 +1,343 @@
 #include "path.hpp"
 
-#include <algorithm>
-
 #include "mgp.hpp"
 
-void Path::Create(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
-  mgp::memory = memory;
-  const auto arguments = mgp::List(args);
-  const auto record_factory = mgp::RecordFactory(result);
-  try {
-    auto start_node{arguments[0].ValueNode()};
-    auto relationships{arguments[1].ValueMap()};
+Path::PathHelper::PathHelper(const mgp::List &labels, const mgp::List &relationships, int64_t min_hops,
+                             int64_t max_hops) {
+  ParseLabels(labels);
+  FilterLabelBoolStatus();
+  ParseRelationships(relationships);
+  config_.min_hops = min_hops;
+  config_.max_hops = max_hops;
+}
 
-    mgp::Path path{start_node};
-    for (const auto &relationship : relationships["rel"].ValueList()) {
-      if (relationship.IsNull()) {
-        break;
-      }
-      if (!relationship.IsRelationship()) {
-        std::ostringstream oss;
-        oss << relationship.Type();
-        throw mgp::ValueException("Expected relationship or null type, got " + oss.str());
-      }
+Path::PathHelper::PathHelper(const mgp::Map &config) {
+  if (!(config.At("minHops").IsNull() || config.At("minHops").IsInt()) ||
+      !(config.At("maxHops").IsNull() || config.At("maxHops").IsInt()) ||
+      !(config.At("relationshipFilter").IsNull() || config.At("relationshipFilter").IsList()) ||
+      !(config.At("labelFilter").IsNull() || config.At("labelFilter").IsList()) ||
+      !(config.At("filterStartNode").IsNull() || config.At("filterStartNode").IsBool()) ||
+      !(config.At("beginSequenceAtStart").IsNull() || config.At("beginSequenceAtStart").IsBool()) ||
+      !(config.At("bfs").IsNull() || config.At("bfs").IsBool())) {
+    throw mgp::ValueException(
+        "The config parameter needs to be a map with keys and values in line with the documentation.");
+  }
```

Comment:
Is it possible to make function which can accept expected types and check for each one that is one of the expected types and throws if it is not

```
if (!config.At("minHops").IsExpectedType({Type::Null, Type::Int})){
    throw mgp::ValueException...
}
```

Reply:
Would this be ok?

`auto same_type_or_null = [] (const mgp::Type type, const mgp::Type wanted_type) {
    return type == wanted_type || type == mgp::Type::Null;
  };

  if (!same_type_or_null(config.At("minHops").Type(), mgp::Type::Int) ||
     !same_type_or_null(config.At("maxHops").Type(), mgp::Type::Int) ||
     !same_type_or_null(config.At("relationshipFilter").Type(), mgp::Type::List) ||
     !same_type_or_null(config.At("labelFilter").Type(), mgp::Type::List) ||
     !same_type_or_null(config.At("filterStartNode").Type(), mgp::Type::Bool) ||
     !same_type_or_null(config.At("beginSequenceAtStart").Type(), mgp::Type::Bool) ||
     !same_type_or_null(config.At("bfs").Type(), mgp::Type::Bool)){
    throw mgp::ValueException(
        "The config parameter needs to be a map with keys and values in line with the documentation.");
     }`

Reply:
Yes, works!